### PR TITLE
List

### PR DESCRIPTION
**Pull request by kuritika**
Inizialised the class variables: please check

### CODE REVIEW

**Review by kuritika — COMMENTED** [review]
Added addrecipe and initialised first attribute: please check

File: src/main/java/de/tastylabs/list/Node.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.util.List;
+
+public class Node extends ListElement{
+    private final ListElement next;
+    private final Recipe recipe;
+
+    public Node(ListElement next, Recipe recipe) {
+        // TODO: Initialise the class variables
+        next = newNext;
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/Node.java:14]:
```suggestion
        this.next = next;
```

**Review by kuritika — COMMENTED** [review]
Add recipe to the Result: Please check

File: src/main/java/de/tastylabs/list/Node.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        if(Recipe=query){
+            System.out.println()
+        }
+
+        List<Recipe> result;
+        // First, get the result of the search on the next element. (result = ...)
+        // Then, check for each word in the query, if either the id, the title, the ingredients or the preparation contain this word. The logical or is in Java || . If so, add it to the result list.
+        for (String word : query) {
+            if (true /* expression here */ ) {
+                // Add the recipe to the result. Use the add method of the result list.
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/Node.java:32]:
```suggestion
            if (recipe.getId().contains(word) || recipe.getTitle().contains(word) || recipe.getIngredients().contains(word) || recipe.getPreparation().contains(word)) {
                result.add(recipe);
```

File: src/main/java/de/tastylabs/list/Node.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    Recipe get(String id) throws RecipeNotFoundException {
+        // TODO: Find the recipe with the id
+        // Check if the id of the recipe equals (!) the id of the parameter
+        // To test if a string is equal to another, please use the equals method.
+        // If the recipe has not the equal id, return the result of the method call on the next list element
+        if (true /* expression here */ ) {
+            return null;
+        } else {
+            return null;
+        }
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/Node.java:48]:
```suggestion
        if (recipe.getId().equals(id)) {
            return recipe;
        } else {
            return next.get(id);
        }
```

File: src/main/java/de/tastylabs/list/Node.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    Recipe get(int index) {
+        // TODO: Find the recipe with this index
+        // Return the recipe if the index is 0
+        // Otherwise, return the result of the method call on the next list element, whereby the index is reduced by 1
+        if (true /* expression here */ ) {
+            return null;
+        } else {
+            return null;
+        }
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/Node.java:60]:
```suggestion
        if (index == 0) {
            return recipe;
        } else {
            return next.get(index -1);
        }
```

File: src/main/java/de/tastylabs/list/Node.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Override
+    int size() {
+        // TODO: Return the size of the list
+        // This is always the size of the following element plus 1
+        return 0;
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/Node.java:67]:
```suggestion
        return next.size() + 1;
```

File: src/main/java/de/tastylabs/list/RecipeList.java [hunk 6]
```diff
@@ -1,5 +1,58 @@
...
+
+import java.util.List;
+
 public class RecipeList {
-    // TODO: List
+    ListElement first;
+
+    public RecipeList() {
+        // TODO: Initialise the first attribute
+      ListElement = newListelement;
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/RecipeList.java:13]:
```suggestion
        first = new Completion();
```

File: src/main/java/de/tastylabs/list/RecipeList.java [hunk 7]
```diff
@@ -1,5 +1,58 @@
...
+    }
+
+    public void add(Recipe recipe) {
+        // TODO: Add new recipe to the list
+        recipe = new Recipe
+
+        Recipe.add(recipe);
+
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/RecipeList.java:21]:
```suggestion
        first = new Node(first, recipe);
```

File: src/main/java/de/tastylabs/list/RecipeList.java [hunk 8]
```diff
@@ -1,5 +1,58 @@
...
+        recipe = new Recipe
+
+        Recipe.add(recipe);
+        
+    }
+
+    public List<Recipe> search(List<String> query) {
+        // TODO: Return the result of the method call on the first list element
+        get ListElement;
+        return;
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/RecipeList.java:27]: Try it with `first`. You have just to call the method on this object.

File: src/main/java/de/tastylabs/list/RecipeList.java [hunk 9]
```diff
@@ -1,5 +1,58 @@
...
+        // TODO: Return a random recipe
+        // Tip: To get a random index use: (int) (Math.random() * size())
+        // Math.random() will generate a random number between 0 and 1, which is multiplied with the size a random index.
+        // (int) means, that the double is then casted to a int value.
+        return null;
+    }
+
+    public int size() {
+        // TODO: Return the result of the method call on the first list element
+        return 0;
```
Comment (bensofficial) [inline src/main/java/de/tastylabs/list/RecipeList.java:56]: Try it with `first`. You have just to call the method on this object.

**Review by kuritika — COMMENTED** [review]
Implemented recipe get: please check

**Pull request merged**